docs/whitepapers/kalmanfilter/kalmanfilter.tex

```diff
@@ -0,0 +1,345 @@
+\documentclass{article}
+\usepackage{amsmath}
+\usepackage[margin=1.2in]{geometry} 
+
+\title{Kalman Filter for Quadcopter Position Hold}
+\author{Peter Davidson and Pantelis Sopasakis}
+
+
+\begin{document}
+\maketitle
+
+\section{Problem statement}
+The system dynamics is described by Newton's second law of motion.
+Let $z$ be the altitude of the quadcopter, $g$ denotes the acceleration
+due to gravity and $a$ is the acceleration that results from the forces
+of the propellers. Then, the system dynamics is as simple as
+\begin{equation}
+    \ddot{z} = a - g.
+\end{equation}
+The acceleration, $a$, is an affine function of the thrust reference signal
+(which comes from the RC); it is
+\begin{eqnarray}
+    a = \alpha \tau + \beta,
+\end{eqnarray}
+where $\alpha$ and $\beta$ are coefficients to be estimated; we can obtain
+\textit{a priori} estimates offline and update them online using measurements
+while flying (using the Kalman filter).
+
+
+@Peter,
+\begin{enumerate}
+    \item Copy here the system dynamics from Chandra's report (we don't need Section 2.2.3)
+    \item Write down what sensors we use and what their characteristics are (level of noise,
```

Comment:
To measure the level of noise, presense of outliers and the bias of the sensor, shouldnt we fly it outside record the values and then come to a conclusion?

Reply:
Exactly. There are some obvious fact, such as:
- all sensors are subjec to noise
- the ToF sensor will have occasional missing values and outliers (as you said, we need to record values)
- the ToF sensor measure distance from the ground
- the GPS gives an absolute altitude wrt to sea level, so the measurements will be significantly different from the ToF
- the ToF is affected by the shape of the terrain
- The barometer, again, gives an estimate of absolute altitude, which, however, will not match that of the GPS
- The barometer has the least precision
- They all run at different frequencies
- A thing to keep in mind is that accelerometer information may be useful